src/org/thoughtcrime/securesms/BaseConversationItem.java

```diff
@@ -0,0 +1,125 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+import android.content.Intent;
+import android.text.util.Linkify;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+
+import com.b44t.messenger.DcChat;
+import com.b44t.messenger.DcContext;
+import com.b44t.messenger.DcMsg;
+
+import org.thoughtcrime.securesms.connect.ApplicationDcContext;
+import org.thoughtcrime.securesms.connect.DcHelper;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public abstract class BaseConversationItem extends LinearLayout
```

Comment:
is that BaseConversationItem from the Signal upstream? not saying that it needs to be, just curious :) if not, maybe worth checking what they are doing, the must have similar problems.

Reply:
@r10s no, I could have just added the needed stuff to ConversationUpdateItem, but then things would be duplicated so I moved things to this base class, stuff from Signal usually have copyright and author notices everywhere :D maybe I should have add something like that so it is clear who to blame :D

Reply:
wrt Signal, just checked, while trying to remove someone from a group while offline, they just don't add the system message until the person is actually removed, and they don't have a contact request "catch all" chat like us